sympy/physics/mechanics/kane.py

```diff
@@ -115,14 +124,50 @@ class KanesMethod(object):
     """
 
     def __init__(self, frame, q_ind, u_ind, kd_eqs=None, q_dependent=None,
-            configuration_constraints=None, u_dependent=None,
-            velocity_constraints=None, acceleration_constraints=None,
-            u_auxiliary=None):
+                 configuration_constraints=None, u_dependent=None,
+                 velocity_constraints=None, acceleration_constraints=None,
+                 u_auxiliary=None):
+        """
+        n : number of independent generalized coordinates
+        m : number of dependent generalized speeds and non-holonomic constraints
+        o : number of dependent generalized coordinates and holonomic constraints
+        p : number of independent generalized speeds
+
+        Parameters
+        ==========
+        frame : ReferenceFrame
+            The inertial reference frame with which to form the equations of
+            motion with respect to. All bodies and particles must have their
+            velocity defined with respect to this frame, either directly or
+            indirectly.
+        q_ind : ordered iterable of SymPy functions of time, len(n)
+            The n independent coordinates.
+        u_ind : ordered iterable, len(p)
+            The p independent generalized speeds.
+        kd_eqs : iterable of SymPy expressions, optional
+            The left hand side of the zero equal kinematical differential
+            equations. If None, the simple definition of u = q' is used.
+        q_dependent : ordered iterable, len(o)
+            The
```

Comment:
Yes, I'm still not done with the PR. Been pecking away at it when I have a few minutes.